homework done

v-surg opened the pull request.

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,51 @@
+    FROM vacancy AS v 
+    LEFT JOIN area AS a ON v.area_id = a.area_id
+    LEFT JOIN employer AS er ON v.employer_id = er.employer_id 
+    WHERE (v.compensation_from IS NULL AND v.compensation_to IS NULL)
```
можно и без скобок

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,51 @@
+-- Task 3
+SELECT v.name AS vacancy, a.name AS city, er.name AS employer 
+    FROM vacancy AS v 
```
у FROM, JOIN, WHERE лишний таб

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,51 @@
+SELECT COUNT(r.response_id) AS responses_per_employer, er.name AS employer
+    FROM response AS r 
+    LEFT JOIN vacancy AS v ON r.vacancy_id = v.vacancy_id
+    LEFT JOIN employer AS er ON v.employer_id = er.employer_id
```
если у работодателя не было вакансий или откликов, его не будет в выборке, надо поправить, чтобы был

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,51 @@
+SELECT PERCENTILE_CONT(0.5) WITHIN GROUP (ORDER BY vac_per_er) AS median_vacancies_per_employer
+    FROM (SELECT count(v.vacancy_id) AS vac_per_er
+        FROM vacancy AS v
+        LEFT JOIN employer AS er ON v.employer_id = er.employer_id
```
работодатели без вакансий потеряются, а для медианы их нули очень важны

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,51 @@
+
+--Task 7
+SELECT MIN(first_resp - vac_created) AS min_lag, MAX(first_resp - vac_created) AS max_lag, a.name AS city
+   FROM (
```
если вынести здесь и в предыдущем запросе вложенный запрос в WITH, то будет более читаемо

v-surg commented:
1. some minor changes
2. to prevent loosing employers in task 5, replaced 'LEFT JOIN employers' with 'FULL OUTER JOIN employers'
3. also changed to 'WITH statement' in task 6

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,53 @@
+SELECT 
+    AVG(CASE WHEN compensation_gross = true THEN compensation_from ELSE compensation_from/0.87 END) AS avg_min_salary, 
+    AVG(CASE WHEN compensation_gross = true THEN compensation_to ELSE compensation_to/0.87 END) AS avg_max_salary,
+    AVG((CASE WHEN compensation_gross = true THEN compensation_from ELSE compensation_from/0.87 END) + 
```
похоже, что тут лишняя строчка

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,53 @@
+SELECT COUNT(r.response_id) AS responses_per_employer, er.name AS employer
+FROM response AS r 
+LEFT JOIN vacancy AS v ON r.vacancy_id = v.vacancy_id
+FULL OUTER JOIN employer AS er ON v.employer_id = er.employer_id
```
OUTER можно не писать

eldmitrieva replied:
и тут было бы достаточно RIGHT JOIN

eldmitrieva commented on make_requests.sql:
```diff
@@ -0,0 +1,53 @@
+WITH subquery AS (
+    SELECT count(v.vacancy_id) AS vac_per_er
+        FROM vacancy AS v
+        LEFT JOIN employer AS er ON v.employer_id = er.employer_id
```
теряются работодатели, у которых не было вакансий
